dashboard/client/src/pages/node/NodeRow.tsx

```diff
@@ -0,0 +1,272 @@
+import { IconButton, TableCell, TableRow, Tooltip } from "@material-ui/core";
+import { createStyles, makeStyles } from "@material-ui/core/styles";
+import AddIcon from "@material-ui/icons/Add";
+import RemoveIcon from "@material-ui/icons/Remove";
+import { sortBy } from "lodash";
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { Link } from "react-router-dom";
+import PercentageBar from "../../components/PercentageBar";
+import { StatusChip } from "../../components/StatusChip";
+import { getNodeDetail } from "../../service/node";
+import { NodeDetail } from "../../type/node";
+import { Worker } from "../../type/worker";
+import { memoryConverter } from "../../util/converter";
+
+const useNodeRowStyles = makeStyles((theme) =>
+  createStyles({
+    expandCollapseIcon: {
+      color: theme.palette.text.secondary,
+      fontSize: "1.5em",
+      verticalAlign: "middle",
+    },
+    idCol: {
+      display: "block",
+      width: "50px",
+      overflow: "hidden",
+      textOverflow: "ellipsis",
+      whiteSpace: "nowrap",
+    },
+  }),
+);
+
+type NodeRowProps = Pick<NodeRowsProps, "node" | "rowIndex"> & {
+  /**
+   * Whether the node has been expanded to show workers
+   */
+  expanded: boolean;
+  /**
+   * Click handler for when one clicks on the expand/unexpand button in this row.
+   */
+  onExpandButtonClick: () => void;
+};
+
+/**
+ * A single row that represents the node information only.
+ * Does not show any data about the node's workers.
+ */
+const NodeRow = ({
+  node,
+  rowIndex,
+  expanded,
+  onExpandButtonClick,
+}: NodeRowProps) => {
+  const {
+    hostname = "",
+    ip = "",
+    cpu = 0,
+    mem,
+    disk,
+    networkSpeed = [0, 0],
+    raylet,
+    logUrl,
+  } = node;
+
+  const classes = useNodeRowStyles();
+
+  const objectStoreTotalMemory =
+    raylet.objectStoreAvailableMemory + raylet.objectStoreUsedMemory;
+
+  return (
+    <TableRow key={hostname + rowIndex}>
```

Comment:
don‘t need the "key" parameter here, but you need it in the component which dashboard/client/src/pages/node/index.tsx and line 280.